spras/btb.py

```diff
@@ -0,0 +1,188 @@
+            except Exception as err:
+                print("error!!")
+                print(err)
+                raise(err)
+
+        work_dir = '/btb'
+
+        # Each volume is a tuple (src, dest)
+        volumes = list()
+
+        bind_path, source_file = prepare_volume(sources, work_dir)
+        volumes.append(bind_path)
+
+        bind_path, target_file = prepare_volume(targets, work_dir)
+        volumes.append(bind_path)
+
+        bind_path, edges_file = prepare_volume(edges, work_dir)
+        volumes.append(bind_path)
+
+        # PathLinker does not provide an argument to set the output directory
+        # Use its --output argument to set the output file prefix to specify an absolute path and prefix
+        out_dir = Path(output_file).parent
+        # PathLinker requires that the output directory exist
+        out_dir.mkdir(parents=True, exist_ok=True)
+        bind_path, mapped_out_dir = prepare_volume(str(out_dir), work_dir)
+        volumes.append(bind_path)
+        mapped_out_prefix = mapped_out_dir + '/raw-pathway.txt'  # Use posix path inside the container
+
+        command = ['python',
+                   'btb.py',
+                   '--edges',
+                   edges_file,
+                   '--sources',
+                   source_file,
+                   '--target',
+                   target_file,
+                   '--output',
+                   mapped_out_prefix]
+        # command = ['ls', '-R']
+
+
+        print('Running BowtieBuilder with arguments: {}'.format(' '.join(command)), flush=True)
+
+        container_suffix = "bowtiebuilder"
+        out = run_container(container_framework,
+                            container_suffix,
+                            command,
+                            volumes,
+                            work_dir)
+        print(out)
+        print("Source file: ", source_file)
+        print("target file: ", target_file)
+        print("edges file: ", edges_file)
+        print("mapped out dir: ", mapped_out_dir)
+        print("mapped out prefix: ", mapped_out_prefix)
+
+
+
+
+
+        # Rename the primary output file to match the desired output filename
+        # Currently PathLinker only writes one output file so we do not need to delete others
+        # We may not know the value of k that was used
+        # output_edges = Path(next(out_dir.glob('out*-ranked-edges.txt')))
+        # output_edges.rename(output_file)
+
+
+    @staticmethod
+    def parse_output(raw_pathway_file, standardized_pathway_file):
+        """
+        Convert a predicted pathway into the universal format
+        @param raw_pathway_file: pathway file produced by an algorithm's run function
+        @param standardized_pathway_file: the same pathway written in the universal format
+        """
+        # What about multiple raw_pathway_files
+        print("PARSING OUTPUT BTB")
+        df = pd.read_csv(raw_pathway_file, sep='\t')
+        # df = reinsert_direction_col_directed(df)
```

Comment:
I think we need to reinsert the direction column (add a `D` column, since the output is directed).

Reply:
I think uncommenting line 186 is what you want.